controller/cmd/service-mirror/main.go

```diff
@@ -44,11 +69,20 @@ func Main(args []string) {
 		log.Fatalf("Failed to initialize K8s API: %s", err)
 	}
 
+	secretsInformer, err := initLocalSecretsInformer(k8sAPI.Client, *namespace)
+
+	if err != nil {
+		log.Fatalf("Failed to initialize secrets informer: %s", err)
+	}
+
 	probeManager := NewProbeManager(k8sAPI)
 	probeManager.Start()
 
-	k8sAPI.Sync(nil)
-	watcher := NewRemoteClusterConfigWatcher(k8sAPI, *requeueLimit, &chanProbeEventSink{probeManager.enqueueEvent})
+	if err := k8sAPI.Sync(nil); err != nil {
+		log.Fatalf("Failed to sync caches for k8s api. Check your service mirror RBACs")
```

Comment:
Without looking at `err` here, the RBAC thing is just a guess.  What if this fails for some other reason?  

Can we actively check RBACs before syncing by doing a `can-i` call?